src/awkward/_v2/cpp-headers/awkward/GrowableBuffer.h

```diff
@@ -184,8 +285,8 @@ namespace awkward {
     /// #reserved, a new panel will be allocated.
     void
     append(PRIMITIVE datum) {
-      if (length_[ptr_.size()-1] == reserved_[ptr_.size()-1]) {
-        add_panel(reserved_[ptr_.size()-1]);
+      if (ptr_->length() == ptr_->reserved()) {
+        add_panel((size_t)ceil(ptr_->reserved() * 1.5));
```

Comment:
The panels grow as they're added? That's not a bad idea: at first, you don't know how big they need to be, but if you've been adding a lot of panels, you can be more confident that large panels are better.

However, if that's the case, then it needs to be possible to parameterize the `resize`, not just fix it as `1.5`. The old algorithm's `resize=1.5` came from the fact that it was a replacement algorithm, and there's some mathematical reason why the golden ratio is the best resize factor (for defragmentation). Adding panels and not removing the old ones is a different dynamic—there's less fragmentation, for one thing—so the best factor might not be `1.5`. Eventually, you'll want to do a performance test, and that would involve varying these two parameters, `initial` and `resize`.

Those are the two parameters that the old `ArrayBuilderOptions` carried, and this PR converts a lot of `options` (which can be extended by modifying `ArrayBuilderOptions`) into `initial` (which can't be extended). If we're already wanting to reintroduce a `factor`, are you sure you want to do this?

On the other hand, I don't yet know that the optimal `resize` isn't `1.0`. Maybe you won't be needing it. But having an `ArrayBuilderOptions`, or at least a "`BuilderOptions`", could be more future-proof.

Reply:
I think, the `resize` parameter should not be set by a user. Also, perhaps, it should not be a fixed number. I agree, we should profile this first. `GrowableBuffer` can have a "growable strategy" hint, for example. I'll merge it as is - this will help @ManasviGoyal to rebase her PR.